apps/web/modules/survey/templates/components/template-container.test.tsx

```diff
@@ -0,0 +1,197 @@
+  SearchBar: vi.fn(({ placeholder, onChange, value }) => (
+    <input
+      data-testid="search-bar"
+      placeholder={placeholder}
+      onChange={(e) => onChange(e.target.value)}
+      value={value}
+    />
+  )),
+}));
+
+vi.mock("@tolgee/react", () => ({
+  useTranslate: vi.fn(() => ({
+    t: (key: string) => {
+      const translations: Record<string, string> = {
+        "environments.surveys.templates.create_a_new_survey": "Create a new survey",
+        "environments.surveys.all_set_time_to_create_first_survey":
+          "All set! Time to create your first survey",
+        "common.search": "Search",
+      };
+      return translations[key] || key;
+    },
+  })),
+}));
+
+vi.mock("../lib/minimal-survey", () => ({
+  getMinimalSurvey: vi.fn(() => ({})),
+}));
+
+const mockProject = {
+  id: "project1",
+  name: "Test Project",
+  createdAt: new Date(),
+  updatedAt: new Date(),
+  organizationId: "org1",
+  config: {
+    channel: "website" as TProjectConfigChannel,
+    industry: "technology" as TProjectConfigIndustry,
+  },
+} as Project;
+
+const mockEnvironment = {
+  id: "env1",
+  appSetupCompleted: true,
+};
+
+const mockPrefilledFilters: (TProjectConfigChannel | TProjectConfigIndustry | TTemplateRole | null)[] = [];
+
+describe("TemplateContainerWithPreview", () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  test("renders MenuBar when isTemplatePage is true", () => {
+    render(
+      <TemplateContainerWithPreview
+        project={mockProject}
+        environment={mockEnvironment}
+        userId="user1"
+        prefilledFilters={mockPrefilledFilters}
+        isTemplatePage={true}
+      />
+    );
+
+    expect(screen.getByTestId("menu-bar")).toBeInTheDocument();
+  });
+
+  test("does not render MenuBar when isTemplatePage is false", () => {
+    render(
+      <TemplateContainerWithPreview
+        project={mockProject}
+        environment={mockEnvironment}
+        userId="user1"
+        prefilledFilters={mockPrefilledFilters}
+        isTemplatePage={false}
+      />
+    );
+
+    expect(screen.queryByTestId("menu-bar")).not.toBeInTheDocument();
+  });
+
+  test("displays correct title when isTemplatePage is true", () => {
+    render(
+      <TemplateContainerWithPreview
+        project={mockProject}
+        environment={mockEnvironment}
+        userId="user1"
+        prefilledFilters={mockPrefilledFilters}
+        isTemplatePage={true}
+      />
+    );
+
+    expect(screen.getByText("Create a new survey")).toBeInTheDocument();
+  });
+
+  test("displays correct title when isTemplatePage is false", () => {
+    render(
+      <TemplateContainerWithPreview
+        project={mockProject}
+        environment={mockEnvironment}
+        userId="user1"
+        prefilledFilters={mockPrefilledFilters}
+        isTemplatePage={false}
+      />
+    );
+
+    expect(screen.getByText("All set! Time to create your first survey")).toBeInTheDocument();
```

Comment:
pls either change this to the actual translation key `environments.surveys.templates.create_a_new_survey` like we have in other tests in the project or use a data test id or some other selector, if the text changes in the future the test would fail 🙏